src/main/java/DukeNUS.java

```diff
@@ -0,0 +1,205 @@
+    }
+
+    /**
+     * Directs the input to execute the correct command.
+     *
+     * @param userInputWords The complete sentence the user inputted, often consisting of the command as the first word
+     *                       and its arguments as its subsequent words.
+     */
+    private static void interpretCommand(String[] userInputWords) {
+        switch (userInputWords[0]) {
+        case "list":
+            if (tasks.isEmpty()) {
+                DukeNUSPrinter.printMessage("You have no tasks. ¯\\_(ツ)_/¯");
+            } else {
+                DukeNUSPrinter.listTasks(tasks);
+            }
+            break;
+        case "find":
+            ArrayList<Task> filteredTasks = getFoundTasks(tasks, userInputWords[1]);
+            if (filteredTasks.isEmpty()) {
+                DukeNUSPrinter.printMessage("Tasks containing the term \"" + userInputWords[1] + "\" were not found.");
+            } else {
+                DukeNUSPrinter.showSearchResults(filteredTasks);
+            }
+            break;
+        case "delete":
+            try {
+                deleteTask(parseInt(userInputWords[1]));
+            } catch (IndexOutOfBoundsException exception) {
+                DukeNUSPrinter.printMessage("☹ Error: The index of the task is missing or out of bounds.");
+            } catch (NumberFormatException exception) {
+                DukeNUSPrinter.printMessage("☹ Error: The task index you have entered is not a number.");
+            } catch (NullPointerException exception) {
+                DukeNUSPrinter.printMessage("☹ Error: The task index you have provided returns a null task.");
+            }
+            break;
+        case "mark":
+            try {
+                markTask(parseInt(userInputWords[1]));
+            } catch (IndexOutOfBoundsException exception) {
+                DukeNUSPrinter.printMessage("☹ Error: The index of the task is missing or out of bounds.");
+            } catch (NumberFormatException exception) {
+                DukeNUSPrinter.printMessage("☹ Error: The task index you have entered is not a number.");
+            } catch (NullPointerException exception) {
+                DukeNUSPrinter.printMessage("☹ Error: The task index you have provided returns a null task.");
+            }
+            break;
+        case "unmark":
+            try {
+                unmarkTask(parseInt(userInputWords[1]));
+            } catch (ArrayIndexOutOfBoundsException exception) {
+                DukeNUSPrinter.printMessage("☹ Error: The index of the task is missing or out of bounds.");
+            } catch (NumberFormatException exception) {
+                DukeNUSPrinter.printMessage("☹ Error: The task index you have entered is not a number.");
+            } catch (NullPointerException exception) {
+                DukeNUSPrinter.printMessage("☹ Error: The task index you have provided returns a null task.");
+            }
```

Comment:
Noticed that there are multiple print messages within your try-catch statements. Maybe you could consider having DukeNUSPrinter, or other classes to handle the printing of the error messages!